File: hello_world.py
#import sys
#sys.path.append('/nix/store/py6ff80g1jjlzrdrwq664rlfzim1y9dg-cloudi-2.0.5/lib/cloudi-2.0.5/api/python/')
import traceback
from cloudi import API, terminate_exception

class Task(object):
    def __init__(self):
        self.__api = API(0) # first/only thread == 0

    def run(self):
        try:
            self.__api.subscribe("hello_world/get",
                                 self.__hello_world)
            self.__api.poll()
        except terminate_exception:
            pass
        except:
            traceback.print_exc(file=sys.stderr)

    def __hello_world(self, request_type, name, pattern,
                      request_info, request,
                      timeout, priority, trans_id, pid):
        return b'Hello World!'

if __name__ == '__main__':
    assert API.thread_count() == 1
    task = Task()
    task.run()

File: hello_world.erl
-module(hello_world).
-behaviour(cloudi_service).

%% cloudi_service callbacks
-export([cloudi_service_init/4,
         cloudi_service_handle_request/11,
         cloudi_service_handle_info/3,
         cloudi_service_terminate/3]).

-include_lib("cloudi_core/include/cloudi_logger.hrl").

-record(state,
    {
    }).

cloudi_service_init(_Args, _Prefix, _Timeout, Dispatcher) ->
    cloudi_service:subscribe(Dispatcher, "hello_world/get"),
    {ok, #state{}}.

cloudi_service_handle_request(_RequestType, _Name, _Pattern,
                              _RequestInfo, _Request,
                              _Timeout, _Priority,
                              _TransId, _Pid,
                              #state{} = State, _Dispatcher) ->
    {reply, <<"Hello World!">>, State}.

cloudi_service_handle_info(Request, State, _Dispatcher) ->
    ?LOG_WARN("Unknown info \"~p\"", [Request]),
    {noreply, State}.

cloudi_service_terminate(_Reason, _Timeout, #state{}) ->
    ok.
